render/gui/src/main/java/net/flintmc/render/gui/windowing/WindowRenderer.java

```diff
@@ -19,38 +19,48 @@
 
 package net.flintmc.render.gui.windowing;
 
+import net.flintmc.render.gui.screen.ScreenName;
+
+import java.util.Collection;
+
 /**
  * Renderer for windows.
  */
 public interface WindowRenderer {
 
-  /**
-   * Called when the renderer is added to a window. The OpenGL context in this method is the one
-   * used for the window.
-   */
-  void initialize();
-
-  /**
-   * Determines if this renderer is taking full control of the content of the window. If this
-   * setting is {@code true} on a renderer attached to the main window, the default render logic of
-   * minecraft will be disabled.
-   *
-   * <p>This setting can't change while a renderer is attached! To change it, re-attach the
-   * renderer with this method returning another value.
-   *
-   * @return {@code true} to mark the renderer as intrusive, {@code false} to mark it as cooperative
-   */
-  boolean isIntrusive();
-
-  /**
-   * Called when window needs to be rendered. The OpenGL context in this method is the one used for
-   * the window.
-   */
-  void render();
-
-  /**
-   * Called when the renderer is removed from a window. The OpenGL context in this method is the one
-   * used for the window.
-   */
-  void cleanup();
+    /**
+     * Called when the renderer is added to a window. The OpenGL context in this method is the one
+     * used for the window.
+     */
+    void initialize();
+
+    /**
+     * Determines if this renderer is taking full control of the content of the window.
+     * If this setting is {@code true} the rendering of every minecraft window in {@link WindowRenderer#getIntrusiveScreens()}
+     * will be cancelled when this renderer is registered.
+     *
+     * @return {@code true} to mark the renderer as intrusive, {@code false} to mark it as cooperative
+     */
+    boolean isIntrusive();
```

Comment:
As discussed before, is this method really required? This should (from my understanding) only be `true`, if `getIntrusiveScreens().size() > 0`. To avoid bad implementations, this method should be removed and checking should always be done with aforementioned check

Reply:
NOTE: After clarification, it was decided to keep the method and make it periodically polled to check if results from `getIntrusiveScreens()`should be used